scripts/quests/crystalWar/WOTG_BAS_0_The_Fighting_Fourth.lua

```diff
@@ -0,0 +1,267 @@
+                onTrigger = function(player, npc)
+                    -- TODO: Conditions to not have a different campaign allegiance?
+
+                    local hasLetter  = player:hasKeyItem(xi.keyItem.BLUE_RECOMMENDATION_LETTER)
+                    local hasRations = player:hasKeyItem(xi.keyItem.BATTLE_RATIONS)
+
+                    -- TODO: Verify the check for rations exists on retail
+                    if hasLetter or hasRations then
+                        -- You won't be considered a member of the Fighting Fourth until you take that blue recommendation letter to Bastok...
+                        return quest:event(8)
+                    else
+                        -- Hold it right there. You don't look like one of ours.
+                        return quest:event(7)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [7] = function(player, csid, option, npc)
+                    if option == 0 then
+                        npcUtil.giveKeyItem(player, xi.keyItem.BLUE_RECOMMENDATION_LETTER)
+                    end
+                end,
+            },
+        },
+
+        [xi.zone.BASTOK_MARKETS_S] =
+        {
+            ['Adelbrecht'] =
+            {
+                onTrigger = function(player, npc)
+                    -- TODO: Conditions to not have a different campaign allegiance?
+
+                    if player:hasKeyItem(xi.keyItem.BLUE_RECOMMENDATION_LETTER) then
+                        -- Greetings, civillian. The Seventh Cohors of the Republican Legion's Fourth Division is currently recruiting new troops.
+                        -- Args (from caps): With recommendation letter
+                        return quest:progressEvent(139, 0, 0)
+                    else
+                        -- Greetings, civillian. The Seventh Cohors of the Republican Legion's Fourth Division is currently recruiting new troops.
+                        -- Args (from caps): Without recommendation letter (after first prompt)
+                        return quest:progressEvent(139, 1, 23)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [139] = function(player, csid, option, npc)
+                    if option == 1 then
+                        player:delKeyItem(xi.keyItem.BLUE_RECOMMENDATION_LETTER)
+                        npcUtil.giveKeyItem(player, xi.keyItem.BATTLE_RATIONS)
+                        quest:begin(player)
+                    end
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_ACCEPTED
+        end,
+
+        [xi.zone.BASTOK_MARKETS_S] =
+        {
+            ['Adelbrecht'] =
+            {
+                onTrigger = function(player, npc)
+                    local prog = quest:getVar(player, 'Prog')
+
+                    if prog == 0 then
```

Comment:
prog 0,1,2 are most likly :event() so they dont block the mission npc is involved with as if you dont click the correct option will continue this cs over and over